### PR TITLE
Project fetch missing libraries

### PR DESCRIPTION
**Pull request by brian-sinquin**
Additional feature to #1422 

> c3c project fetch

Search for dependencies used in the project file and fetch those that are not present in the project tree.
Works both on root dependencies and target specific dependencies.

### CODE REVIEW

**Comment by brian-sinquin** [comment]
Done.
It should compile on every machine and work only on windows for the moment because of cURL.

File: src/build/build_options.c [hunk 1]
```diff
@@ -187,6 +188,7 @@ static void project_usage()
 	PRINTF("Project Subcommands:");
 	PRINTF("  view                                          view the current projects structure");
 	PRINTF("  add-target <name>  <target_type>              add a new target to the project");
+	PRINTF("  fetch              							fetch missing project libraries");
```
Comment (lerno) [inline src/build/build_options.c:191]: This should only be visible on `FETCH_AVAILABLE` right?

Comment (brian-sinquin) [reply src/build/build_options.c:191]: Yep

Comment (lerno) [reply src/build/build_options.c:191]: But it looks like it's printed even if FETCH_AVAILABLE isn't set?

File: src/build/project_manipulation.c [hunk 2]
```diff
@@ -252,6 +267,95 @@ static void view_target(const char *filename, const char *name, JSONObject *targ
...
+
+	// dependency check tree
+	while (vec_size(deps) > 0)
+	{
+	
+		FOREACH(const char*, dir, libdirs)
+		{
+			
+			const char* dep = deps[vec_size(deps)-1];
+			if(file_exists(file_append_path(dir, str_printf("%s.c3l", dep))))
```
Comment (lerno) [inline src/build/project_manipulation.c:326]: Nitpicking: space between `if` and `(`

**Comment by lerno** [comment]
Oh, and please add this to the release notes as well, including a not of the other feature you implemented.

File: src/build/build_options.c (hunk 1 above)
Comment (brian-sinquin) [reply src/build/build_options.c:191]: Yes I forgot this, I'll fix it in the next hours

**Comment by brian-sinquin** [comment]
> Oh, and please add this to the release notes as well, including a not of the other feature you implemented.

Done

**Comment by lerno** [comment]
Thank you!

**Pull request merged**